app/jobs/update_allele_registry_ids.rb

```diff
@@ -1,72 +1,34 @@
 require 'rbconfig'
 
-class UpdateAlleleRegistryIds < ActiveJob::Base
+class UpdateAlleleRegistryIds < AlleleRegistryIds
   def perform(recurring = true)
     begin
-      for v in Variant.where(allele_registry_id: nil) do
-        allele_registry_id = get_allele_registry_id(v)
-        v.allele_registry_id = allele_registry_id
-        v.save
-      end
       for v in Variant.where.not(allele_registry_id: nil) do
+        old_allele_registry_id = v.allele_registry_id
         allele_registry_id = get_allele_registry_id(v)
-        add_allele_registry_link(allele_registry_id)
+        if allele_registry_id != old_allele_registry_id
+          v.allele_registry_id = allele_registry_id
+          v.save
+          add_allele_registry_link(allele_registry_id)
+          #delete the linkout if no other variant has this allele registry ID
+          if Variant.find_by(allele_registry_id: old_allele_registry_id).blank?
```

Comment:
You could use `exists?` here instead of `blank?` which will do a `SQL` exists query instead of actually loading an object if it finds one.